exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DefaultSqlHandler.java

```diff
@@ -277,6 +277,10 @@ protected DrillRel convertToRawDrel(final RelNode relNode) throws SqlUnsupported
         }
       }
 
+      if (context.getPlannerSettings().isSemiJoinEnabled() &&
```

Comment:
Pls add a brief explanation as comments for why the semi join conversion is done at this stage of the planning process after the normal join ordering is done.